src/Nancy/NancyEngine.cs

```diff
@@ -83,7 +83,7 @@ public class NancyEngine : INancyEngine
 
             this.dispatcher = dispatcher;
             this.contextFactory = contextFactory;
-            this.statusCodeHandlers = statusCodeHandlers;
+            this.statusCodeHandlers = statusCodeHandlers.ToArray();

```

Comment:
Calling `ToArray()` will copy the list passed in, can't we just make the argument array to begin with and avoid the copy?